rmw_implementation/src/functions.cpp

```diff
@@ -36,95 +31,27 @@
 #include "rmw/get_topic_names_and_types.h"
 #include "rmw/rmw.h"
 
-std::string get_env_var(const char * env_var)
-{
-  char * value = nullptr;
-#ifndef _WIN32
-  value = getenv(env_var);
-#else
-  size_t value_size;
-  _dupenv_s(&value, &value_size, env_var);
-#endif
-  std::string value_str = "";
-  if (value) {
-    value_str = value;
-#ifdef _WIN32
-    free(value);
-#endif
-  }
-  // printf("get_env_var(%s) = %s\n", env_var, value_str.c_str());
-  return value_str;
-}
-
-std::list<std::string> split(const std::string & value, const char delimiter)
-{
-  std::list<std::string> list;
-  std::istringstream ss(value);
-  std::string s;
-  while (std::getline(ss, s, delimiter)) {
-    list.push_back(s);
-  }
-  // printf("split(%s) = %zu\n", value.c_str(), list.size());
-  return list;
-}
-
-bool is_file_exist(const char * filename)
-{
-  std::ifstream h(filename);
-  // printf("is_file_exist(%s) = %s\n", filename, h.good() ? "true" : "false");
-  return h.good();
-}
-
-std::string find_library_path(const std::string & library_name)
-{
-  const char * env_var;
-  char separator;
-  const char * filename_prefix;
-  const char * filename_extension;
-#ifdef _WIN32
-  env_var = "PATH";
-  separator = ';';
-  filename_prefix = "";
-  filename_extension = ".dll";
-#elif __APPLE__
-  env_var = "DYLD_LIBRARY_PATH";
-  separator = ':';
-  filename_prefix = "lib";
-  filename_extension = ".dylib";
-#else
-  env_var = "LD_LIBRARY_PATH";
-  separator = ':';
-  filename_prefix = "lib";
-  filename_extension = ".so";
-#endif
-  std::string search_path = get_env_var(env_var);
-  std::list<std::string> search_paths = split(search_path, separator);
-
-  std::string filename = filename_prefix;
-  filename += library_name + filename_extension;
-
-  for (auto it : search_paths) {
-    std::string path = it + "/" + filename;
-    if (is_file_exist(path.c_str())) {
-      return path;
-    }
-  }
-  return "";
-}
-
 #define STRINGIFY_(s) #s
 #define STRINGIFY(s) STRINGIFY_(s)
 
+namespace {
+
 Poco::SharedLibrary *
 get_library()
 {
   static Poco::SharedLibrary * lib = nullptr;
   if (!lib) {
-    std::string env_var = get_env_var("RMW_IMPLEMENTATION");
+    const char * cenv_var{};
+    const char * cenv_err = rcutils_get_env("RMW_IMPLEMENTATION", &cenv_var);
```

Comment:
Missing #include for this function: see failing CI builds.

Reply:
@EricCousineau-TRI Please make sure that the patches build locally for you.

Reply:
Done. Built locally using base ros2 source workspace, `colcon build --packages-up-to rmw_implementation`, and then confirmed testing with `colcon test --packages-select rmw_implementation` (though I couldn't figure out verbose output, so manually ran `ctest -V -R`).

Reply:
Er, I actually don't know why it's failing on CI for `rmw_publisher_options_t`... I just built off of most packages using the `master` branch. Did I update too far?

Reply:
> I couldn't figure out verbose output

You can let `build` / `test` output the content using `--event-handlers console_cohesion+` or check the log files afterwards (in `log/latest/<pkgname>/*`). You can also get the test output for failed tests using `colcon test-result --verbose` after running the tests.

> I actually don't know why it's failing on CI for `rmw_publisher_options_t`...

The PR build is expected to fail since the target branch requires the latest development branch of its dependencies. Those haven't been released yet which is why that job fails.

That is why we run the CI jobs on ci.ros2.org manually to check for a consistent state across the development branches of multiple repositories:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=8321)](http://ci.ros2.org/job/ci_linux/8321/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=6784)](http://ci.ros2.org/job/ci_osx/6784/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8221)](http://ci.ros2.org/job/ci_windows/8221/)